### PR TITLE
metadata tests

### PR DESCRIPTION
**Pull request by kim-tsao**
### What does this PR do?
Adds test coverage for the Metadata object.  Corresponding changes in the library repo will be added once this PR has been approved and merged.  In addition, some Kube and openshift tests were overlooked in a previous PR and added.

### What issues does this PR fix or reference?
https://github.com/devfile/api/issues/502

### Is your PR tested? Consider putting some instruction how to test your changes
Yes, I ran the tests locally and inspected the generated yaml files and test.log for expected output

#### Docs PR
N/A

### CODE REVIEW

File: test/v200/utils/common/metadata_test_utils.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		numTags := GetRandomNumber(1, 5)
+		LogInfoMessage(fmt.Sprintf("   ....... add %d tag(s) to tags ", numTags))
+		for i := 0; i < numTags; i++ {
+			metadata.Tags = append(metadata.Tags, GetRandomString(8, false))
+		}
+
+	}
+
+	if GetRandomDecision(2, 1) {
+		version, err := reggen.Generate("^([0-9]+)\\.([0-9]+)\\.([0-9]+)(\\-[0-9a-z-]+(\\.[0-9a-z-]+)*)?(\\+[0-9A-Za-z-]+(\\.[0-9A-Za-z-]+)*)?$", 3)
```
Comment (yangcao77) [inline test/v200/utils/common/metadata_test_utils.go:59]: add a comment for regex format with an example

**Review by yangcao77 — COMMENTED** [review]
Can you add comment for each  function in `metadata_test_utils.go`?

File: test/v200/utils/common/metadata_test_utils.go (hunk 1 above)
Comment (kim-tsao) [reply test/v200/utils/common/metadata_test_utils.go:65]: comment added

**Comment by kim-tsao** [comment]
> Can you add comment for each function in `metadata_test_utils.go`?

yes, I just added a comment to each function.

**Review by yangcao77 — APPROVED** [review]
lgtm

**Pull request merged**